aten/src/ATen/native/BackendSelectRegister.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <c10/core/TensorOptions.h>
+
+// TODO:
+// 1) add all factory functions
+// 2) autogen this
+
+namespace at {
+namespace native {
+  Tensor empty_backend(IntArrayRef size, const TensorOptions & options, c10::optional<MemoryFormat> memory_format) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::empty", "memory_format");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, const TensorOptions &, c10::optional<MemoryFormat>>(key, size, options, memory_format);
+  }
+
+  Tensor empty_strided_backend(IntArrayRef size, IntArrayRef stride, const TensorOptions & options) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::empty_strided", "");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, IntArrayRef, const TensorOptions &>(key, size, stride, options);
+  }
+
+  Tensor empty_affine_quantized_backend(IntArrayRef size, const TensorOptions & options, double scale, int64_t zero_point, c10::optional<MemoryFormat> memory_format) {
```

Comment:
I'm a little surprised to see TensorOptions here; I'm guessing this is an intermediate state, but a comment to that effect would be helpful.

Reply:
Once TO are gone, these guys will look something like this: 
`
Tensor full_like(const Tensor & self, Scalar fill_value, c10::optional<c10::ScalarType> dtype, c10::optional<c10::Layout> layout, c10::optional<c10::Device> device, c10::optional<bool> pin_memory, c10::optional<MemoryFormat> memory_format) {
  DispatchKey key = TensorOptions.calculate_dispatch_key(dtype, layout, device);
  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::full_like", "dtype");
  return op.callUnboxedWithDispatchKey<Tensor, const Tensor &, Scalar, c10::optional<c10::ScalarType>, c10::optional<c10::Layout>, c10::optional<c10::Device>, c10::optional<bool>, c10::optional<MemoryFormat>>(key, self, fill_value, dtype, layout, device, pin_memory, memory_format);
}
`